### PR TITLE
check parser output

### PR DESCRIPTION
**Pull request by ltn22**
Add a test for parser. The file coap_icmp.pcap contains a coap message and an ICMP error message. The test_parser now check the results.

### CODE REVIEW

**Comment by ltn22** [comment]
Added a patch to parse correctly the Token

**Comment by MarinoMtz** [comment]
Some errors in the test parser while doing the asserts functions in the test

File: src/tests/test_parser.py [hunk 1]
```diff
@@ -18,15 +42,47 @@ def test_parse():
...
+        direction = 'UP',
+        layers    = ["IPv6", "ICMP", "UDP", "COAP"]
+    )
+
+    assert (parsed_ICMP[2] == None) # no error during parsing
+
+    for fd in parsed_ICMP[0]: # check Field Description
+        expected_fields = EXPECTED_RESULTS_ICMP[0][fd]
+
+        assert (parsed_ICMP[0][fd]) == expected_fields
```
Comment (ltn22) [inline src/tests/test_parser.py:75]: assert (parsed_ICMP[0][fd] == expected_fields)

File: src/tests/test_parser.py [hunk 2]
```diff
@@ -18,15 +42,47 @@ def test_parse():
...
+    assert (parsed_ICMP[1] == EXPECTED_RESULTS_IPv6_UDP[1]) # check data part
```
Comment (ltn22) [inline src/tests/test_parser.py:76]: assert (parsed_ICMP[1] == EXPECTED_RESULTS_ICMP[1]) # check data part

File: src/tests/test_parser.py (hunk 1 above)
Comment (MarinoMtz) [reply src/tests/test_parser.py]: changed

File: src/tests/test_parser.py (hunk 2 above)
Comment (MarinoMtz) [reply src/tests/test_parser.py]: changed

**Review by MarinoMtz — APPROVED** [review]

**Pull request merged**